src/Components/MediaCard/index.js

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+import CardContent from "@mui/material/CardContent";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import { Box } from "@mui/material";
+import styles from "./style.module.css";
+
+export default function MediaCard({
+  image,
+  title,
+  description,
+  click,
+  titleStyles,
+  descriptionStyles,
+  buttonStyles,
+}) {
+  return (
+    <Box>
+      <img src={`./img/${image}`} alt="1" />
+      <CardContent className={styles.text}>
+        <Typography
+          variant="h5"
+          component="div"
+          className={styles.title}
+          style={titleStyles && titleStyles(title)}
```

Comment:
I am against the styling like this. 
Please refer to the sx property of the Typography.
https://mui.com/material-ui/api/typography/
and since you are already using className, why not pass all the styles from the className?